rtos/mbed_boot.c

```diff
@@ -400,7 +400,14 @@ void pre_main (void)
    With the RTOS there is not only one stack above the heap, there are multiple
    stacks and some of them are underneath the heap pointer.
 */
+#if defined (__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050)
+__asm(".global __use_two_region_memory\n\t");
+__asm(".global __use_no_semihosting\n\t");
```

Comment:
Does the ARMCC reference manual recommend setting `__use_two_region_memory` and `__use_no_semihosting` using this syntax? Its a bit odd seeing inline assembly outside the body of a function.